intervaltree/intervaltree.py

```diff
@@ -437,6 +437,17 @@ def intersection_update(self, other):
             if iv not in other:
                 self.remove(iv)
 
+    def overlap_intervals(self,other):
+        """
+        Returns a new IntervalTree consisting of intervals representing the
+        regions overlapped by at least one interval in both of self and other.
+        """
+        splits = (self | other)
+        splits.split_overlaps()
+        self_int_other =  IntervalTree(filter(lambda r: self.overlaps(r) and other.overlaps(r), splits))
+        self_int_other.merge_overlaps()
```

Comment:
Mainly, I was after API compatibility with the R functions that do the same thing (in this case, IRanges intersect() function - unfortunately 'intersect' is already used with different semantics in this library). My own use case for this kind of function always involves merging because I want to do computations on the parts in common between two sets of ranges, and I rarely want to do that computation on the same interval twice, so merging makes sense. I can imagine there could be uses where that's not the case. I tend to port a lot of code b/w R and python so having something that behaves the same way between the two was useful to me.